### PR TITLE
Check for live registration config when no arguments passed

### LINKED ISSUE #434: Elemental hosts should run elemental-register at each boot
After registration and provisioning, each host should re-register at each reboot.
This would ensure:
1. the host is tracked by a MachineInventory in the Elemental Operator
2. labels and annotations (in particular the IP addr annotation, SMBIOS and HW labels) are updated

Note that point 1. could be useful only if the host has not been made part of a cluster yet, otherwise  the tracked state could be a bit tricky: if the host is already part of a cluster (provisioned with a kubernetes deployment) it cannot be completely re-synced.
This case is *not* part of this card, and should be tracked in a follow-up one.

### PR DESCRIPTION
**Pull request by anmazzotti**
Close #434

### CODE REVIEW

**Review by fgiudici — APPROVED** [review]
LGTM

**Review by kkaempf — APPROVED** [review]
:+1:

**Comment by anmazzotti** [comment]
Switching the PR back to draft since I'll add new logic to prevent re-registration of machines after deletion of their inventories.

**Review by fgiudici — CHANGES_REQUESTED** [review]
Nice patch!
The only thing that worries me is adding logic inside the binary preventing  registration of systems not booting from a live ISO. I think that should be changed before merging.
For the rest the patch is pretty good 👍🏼

File: cmd/register/main.go [hunk 1]
```diff
@@ -156,18 +157,38 @@ func run(config elementalv1.Config) {
...
 		}
 
 		log.Debugf("Fetched configuration from manager cluster:\n%s\n\n", string(data))
 
-		if yaml.Unmarshal(data, &config) != nil {
-			log.Error("failed to parse registration configuration: ", err)
+		if err := yaml.Unmarshal(data, &config); err != nil {
+			log.Errorf("failed to parse registration configuration: %w", err)
+			if isRegistrationUpdate {
+				log.Debugln("Registration update failed, will not retry again")
```
[the remark this answers is not in the document]
Comment (anmazzotti) [reply cmd/register/main.go]: Good call. I removed the message since it's indeed misleading. 
We still have the Error message to notify this error.

**Review by fgiudici — APPROVED** [review]
LGTM

**Pull request merged**